### PR TITLE
feat: support `onSelect` callback

### LINKED ISSUE #237: `onSelect` property removed v2.x
Hello,

Comparing the exposed component properties between v1 and v2, I noticed that the `onSelect` property was removed, but couldn't find any reference for this breaking change within the release notes changelog.

Was its removal intentional, and if not, would it be possible for it to be added back? If thats the case, I would be open to make the change and open a PR for it.

Thanks in advance.

### PR DESCRIPTION
**Pull request by mfklauberg**
Adding back the `onSelect` property.

closes #237

### CODE REVIEW

File: src/components/DataTypes/createEasyType.tsx [hunk 1]
```diff
@@ -22,8 +22,10 @@ export function createEasyType<Value> (
   const EasyType: React.FC<DataItemProps<Value>> = (props) => {
     const storeDisplayDataTypes = useJsonViewerStore(store => store.displayDataTypes)
     const color = useJsonViewerStore(store => store.colorspace[colorKey])
+    const onSelect = useJsonViewerStore(store => store.onSelect)
+
     return (
-      <DataBox sx={{ color }} >
+      <DataBox onClick={() => onSelect(props.path, props.value)} sx={{ color }} >
```
Comment (pionxzh) [inline src/components/DataTypes/createEasyType.tsx:28]: `onSelect` might be `undefined`.
```suggestion
      <DataBox onClick={() => onSelect?.(props.path, props.value)} sx={{ color }} >
```

**Comment by pionxzh** [comment]
I'm thinking about moving this logic to `DataKeyPair` so that customized value can also support `onSelect` event (ofc the user needs to implement `onClick=props.onSelect`). But we can merge this as a first version 😄

File: src/components/DataTypes/createEasyType.tsx (hunk 1 above)
Comment (mfklauberg) [reply src/components/DataTypes/createEasyType.tsx]: Fixed

**Comment by mfklauberg** [comment]
> I'm thinking about moving this logic to `DataKeyPair` so that customized value can also support `onSelect` event (ofc the user needs to implement `onClick=props.onSelect`). But we can merge this as a first version 😄

Before opening the issue and the PR, one of my options was to implement a customized values to put a click event in the DOM, so I think that this is already possible today, just the event registered in the main component won't be the same and would need to be used in all custom rendered values.

**Review by pionxzh — APPROVED** [review]
LGTM

**Pull request merged**